# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1254810176 -7200
#      Tue Oct 06 08:22:56 2009 +0200
# Node ID 51c17bd185632c6ba8a60a150ddeeab3c6ce42d7
# Parent  01a1fd9167e05a04809aeb7d871799601af56d17
omission in base-lu::unpack

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2009-10-05  Jaroslav Hajek  <highegg@gmail.com>
+
+	* base-lu.cc (base_lu::unpack): Unpack getp as well.
+
 2009-10-05  Jaroslav Hajek  <highegg@gmail.com>
 
 	* PermMatrix.cc (PermMatrix::determinant): Implement a (hopefully)
 	working algorithm.
 
 2009-10-05  Jaroslav Hajek  <highegg@gmail.com>
 
 	* dim-vector.h (operator ==): Include fast case.
diff --git a/liboctave/base-lu.cc b/liboctave/base-lu.cc
--- a/liboctave/base-lu.cc
+++ b/liboctave/base-lu.cc
@@ -45,16 +45,17 @@ base_lu <lu_type> :: packed (void) const
 template <class lu_type>
 void
 base_lu <lu_type> :: unpack (void)
 {
   if (packed ())
     {
       l_fact = L ();
       a_fact = U (); // FIXME: sub-optimal
+      ipvt = getp ();
     }
 }
 
 template <class lu_type>
 lu_type
 base_lu <lu_type> :: L (void) const
 {
   if (packed ())
